cli/src/main/kotlin/Main.kt

```diff
@@ -0,0 +1,535 @@
+
+        @Parameter(description = "Ignore versions of required tools. NOTE: This may lead to erroneous results.",
+                names = ["--ignore-tool-versions"],
+                order = PARAMETER_ORDER_OPTIONAL)
+        private var ignoreToolVersions = false
+
+        @Parameter(description = "Allow dynamic versions of dependencies. This can result in unstable results when " +
+                "dependencies use version ranges. This option only affects package managers that support lock files, " +
+                "like NPM.",
+                names = ["--allow-dynamic-versions"],
+                order = PARAMETER_ORDER_OPTIONAL)
+        private var allowDynamicVersions = false
+
+        @Parameter(description = "By default the components configured to be excluded in the .ort.yml file of the " +
+                "input directory will still be analyzed, but marked as excluded in the result. With this option " +
+                "enabled they will not be analyzed and not appear in the result.",
+                names = ["--remove-excludes-from-result"],
+                order = PARAMETER_ORDER_OPTIONAL)
+        private var removeExcludesFromResult = false
+
+        @Parameter(description = "A YAML file that contains package curation data.",
+                names = ["--package-curations-file"],
+                order = PARAMETER_ORDER_OPTIONAL)
+        private var packageCurationsFile: File? = null
+
+        override fun run() {
+            val absoluteOutputPath = outputDir.absoluteFile
+            if (absoluteOutputPath.exists()) {
+                log.error { "The output directory '$absoluteOutputPath' must not exist yet." }
+                exitProcess(1)
+            }
+
+            println("The following package managers are activated:")
+            println("\t" + packageManagers.joinToString(", "))
+
+            val absoluteProjectPath = inputDir.absoluteFile
+            println("Scanning project path:\n\t$absoluteProjectPath")
+
+            val config = AnalyzerConfiguration(ignoreToolVersions, allowDynamicVersions, removeExcludesFromResult)
+            val analyzer = Analyzer(config)
+            val ortResult = analyzer.analyze(absoluteProjectPath, packageManagers, packageCurationsFile)
+
+            absoluteOutputPath.safeMkdirs()
+
+            outputFormats.forEach { format ->
+                val outputFile = File(absoluteOutputPath, "analyzer-result.${format.fileExtension}")
+                println("Writing analyzer result to '$outputFile'.")
+                format.mapper.writerWithDefaultPrettyPrinter().writeValue(outputFile, ortResult)
+            }
+        }
+    }
+
+    @Parameters(commandNames = ["download"], commandDescription = "Fetch source code from a remote location.")
+    private object DownloaderCommand : Runnable {
+        @Parameter(description = "A dependencies analysis file to use. Must not be used together with '--project-url'.",
+                names = ["--dependencies-file", "-d"],
```

Comment:
Should be `--analyzer-result` or `--analyzer-result-file`. But maybe should be refactored seperately.

Reply:
Yes, let's do that separately, too. I tried to keep the diff for moved code small to be able to easier check if mistakes were made during the move.